nebula3/gclient/net/SessionPool.py

```diff
@@ -170,34 +170,37 @@ def execute_parameter(self, stmt, params):
         """
         session = self._get_idle_session()
         if session is None:
-            raise RuntimeError('Get session failed')
+            raise RuntimeError("Get session failed")
         self._add_session_to_active(session)
 
         try:
             resp = session.execute_parameter(stmt, params)
 
-            # reset the space name to the pool config
-            if resp.space_name() != self._space_name:
-                self._set_space_to_default(session)
-
-            # move the session back to the idle list
-            self._return_session(session)
-
-            return resp
-        except SessionException as se:
-            if se.type in [
-                SessionException.E_SESSION_INVALID,
-                SessionException.E_SESSION_TIMEOUT,
+            # Check for session validity based on error code
+            if resp.error_code() in [
+                ErrorCode.E_SESSION_INVALID,
+                ErrorCode.E_SESSION_TIMEOUT,
             ]:
                 self._active_sessions.remove(session)
                 session = self._get_idle_session()
                 if session is None:
-                    raise RuntimeError('Get session failed')
+                    logger.warning(
```

Comment:
why just print  warn log when session is None?
self._add_session_to_idle(session) will put a None session.

Reply:
Sorry, silly me, fixed.